### ISSUE TITLE
Project no longer builds in Docker with command `RUN npx quasar build` after 2.12.0 upgrade

### ISSUE THREAD

**Issue by brianoflondon** [quasarframework/quasar#15806]
### What happened?

Prior to this update:

```
bol@Dot v4vapp-frontend-v2 % quasar upgrade -i

 Global Quasar CLI • Gathering information with yarn...

 @quasar/app-webpack: 3.3.3 → 3.9.0
 @quasar/extras: 1.16.2 → 1.16.3
 quasar: 2.11.10 → 2.12.0
 @quasar/app-vite: 1.2.1 → 1.4.0
 ```
 
 The following Dockerfile would successfully build a Quasar project:
 
 ```
 # Stage 1: Build the Quasar project
FROM node:16 AS build

# Set working directory
WORKDIR /app

# Copy package*.json and yarn.lock
COPY package*.json ./
COPY yarn.lock ./

# Install dependencies
RUN yarn install

# Copy the rest of the application code
COPY . .

# Build the Quasar app for production
RUN npx quasar build
```

After the upgrade, the following error occurs during the Docker build:

```
 > [build 7/7] RUN npx quasar build:                                                                                                             
#13 0.415                                                                                                                                        
#13 0.415  .d88888b.                                                                                                                             
#13 0.415 d88P" "Y88b
#13 0.415 888     888
#13 0.415 888     888 888  888  8888b.  .d8888b   8888b.  888d888
#13 0.415 888     888 888  888     "88b 88K          "88b 888P"
#13 0.415 888 Y8b 888 888  888 .d888888 "Y8888b. .d888888 888
#13 0.415 Y88b.Y8b88P Y88b 888 888  888      X88 888  888 888
#13 0.415  "Y888888"   "Y88888 "Y888888  88888P' "Y888888 888
#13 0.415        Y8b
#13 0.415 
#13 0.422 
#13 0.422  Build mode................ spa
#13 0.422  Pkg quasar................ v2.12.0
#13 0.422  Pkg @quasar/app-webpack... v3.9.0
#13 0.422  Pkg webpack............... v5
#13 0.422  Debugging................. no
#13 0.422  Publishing................ no
#13 0.422 
#13 0.426  Configured browser support (>= 88.33% of global marketshare):
#13 0.426  · Chrome for Android >= 111
#13 0.426  · Firefox for Android >= 110
#13 0.426  · QQ for Android >= 13.1
#13 0.426  · UC for Android >= 13.4
#13 0.426  · Android >= 111
#13 0.426  · Chrome >= 109
#13 0.426  · Edge >= 110
#13 0.426  · Firefox >= 102
#13 0.426  · iOS >= 15.6
#13 0.426  · Kaios >= 2.5
#13 0.426  · Opera Mini >= all
#13 0.426  · Opera Mobile >= 73
#13 0.426  · Opera >= 94
#13 0.426  · Safari >= 15.6
#13 0.426  · Samsung >= 19.0
#13 0.426 
#13 0.468 node:fs:590
#13 0.468   handleErrorFromBinding(ctx);
#13 0.468   ^
#13 0.468 
#13 0.468 Error: ENOENT: no such file or directory, open '/app/src/index.template.html'
#13 0.468     at Object.openSync (node:fs:590:3)
#13 0.468     at Object.readFileSync (node:fs:458:35)
#13 0.468     at module.exports (/app/node_modules/@quasar/app-webpack/lib/helpers/app-files-validations.js:10:22)
#13 0.468     at QuasarConfFile.compile (/app/node_modules/@quasar/app-webpack/lib/quasar-conf-file.js:532:5)
#13 0.468     at processTicksAndRejections (node:internal/process/task_queues:96:5)
#13 0.468     at async build (/app/node_modules/@quasar/app-webpack/lib/cmd/build.js:184:3) {
#13 0.468   errno: -2,
#13 0.468   syscall: 'open',
#13 0.468   code: 'ENOENT',
#13 0.468   path: '/app/src/index.template.html'
#13 0.468 }
```

### What did you expect to happen?

Prior to the upgrade or if I change the Dockerfile to this:

```
# Build the Quasar app for production
# RUN npx quasar build
RUN yarn global add @quasar/cli
RUN quasar build
```

The build succeeds:

```
 => CACHED [build 2/8] WORKDIR /app                                                                                                         0.0s
 => CACHED [build 3/8] COPY package*.json ./                                                                                                0.0s
 => CACHED [build 4/8] COPY yarn.lock ./                                                                                                    0.0s
 => CACHED [build 5/8] RUN yarn install                                                                                                     0.0s
 => CACHED [build 6/8] COPY . .                                                                                                             0.0s
 => CACHED [build 7/8] RUN yarn global add @quasar/cli                                                                                      0.0s
 => CACHED [build 8/8] RUN quasar build                                                                                                     0.0s
 => CACHED [stage-1 2/3] COPY --from=build /app/dist/spa /app/dist/spa                                                                      0.0s
 => CACHED [stage-1 3/3] COPY nginx/nginx.conf /etc/nginx/conf.d/default.conf                                                               0.0s
 => exporting to image                                                                                                                      0.0s
 => => exporting layers                                                                                                                     0.0s
 => => writing image sha256:ad8cff2826f69fb7b9145b6bd3f2f4d989f1382152958ad2a2cb3b1c619d5621                                                0.0s
 => => naming to docker.io/library/v4vappfrontendv2:latest  
 ```

### Reproduction URL

https://github.com/brianoflondon/v4vapp-frontend-v2/blob/feature/add-dockerfile/Dockerfile

### How to reproduce?

I'm not sure I can set up a test environment where you can build the Dockerfile. If there is a way, I don't know it.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

SPA Mode

### Quasar info output

```Shell
Operating System - Darwin(22.4.0) - darwin/arm64
NodeJs - 18.15.0

Global packages
  NPM - 9.5.0
  yarn - 1.22.19
  @quasar/cli - 2.1.1
  @quasar/icongenie - 2.5.4
  cordova - Not installed

Important local packages
  quasar - 2.12.0 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.4.0 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.16.3 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.47 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.6
  pinia - 2.0.34 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.38.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - Dot.local
  en0 - 10.0.0.204
  utun3 - 100.89.33.19
```

**Comment by rstoenescu** [comment]
Hi,

Your project seems to be a q/app-vite one, but you have both q/app-vite and q/app-webpack specified in your package.json.
Remove q/app-webpack. Also, for future reference, q/app-* should never be in "dependencies" but in "devDependencies".

![image](https://github.com/quasarframework/quasar/assets/4665641/b168dcbc-335a-4f3c-9588-62b962d2ef09)

**Comment by brianoflondon** [comment]
Thank you so much!

I haven't directly edited the `package.json` myself, I'm just wondering how those conflicts got in there. The only thing I've done with this project is run the installation from the docs:

```
$ yarn global add @quasar/cli
$ yarn create quasar
```

And then a little later I performed the upgrade. Just wondering if there is some issue with the dependencies this specified for me and the best way to remove things.

I presume `yarn remove @quasar/app-webpack` is all I need to do (update... that worked).

**Comment by rstoenescu** [comment]
You 100% had either "@quasar/app" or "@quasar/app-webpack" in the package.json because it has been detected by the "quasar upgrade" command. But just by using "yarn create quasar" it's impossible to get both q/app-*.

In any case, glad it now works for you.

**Comment by brianoflondon** [comment]
The whole package management system in Javascript is new to me and baffling. Not quite sure how I got webpack in there but now I know what to look for.

**Comment by brianoflondon** [comment]
I know you @rstoenescu closed this but I looked at the exact steps I did to an otherwise unchanged project are this:

`yarn upgrade @quasar/app@latest`

That didn't add in webpack but this:

`quasar upgrade -i`

Absolutely did change it.

<img width="1071" alt="image" src="https://github.com/quasarframework/quasar/assets/28054569/a07eda17-484c-483f-bf4b-2e9b50a9e37d">

**Comment by rstoenescu** [comment]
Hi,

Expanding on the explanation. The `@quasar/app` package is the webpack one. I got it renamed to `@quasar/app-webpack` when I introduced q/app-vite to better align with the namings (q/app-webpack and q/app-vite).

What the q/cli "upgrade" command does is it looks for "quasar" and any "@quasar/*" packages and upgrades them. Naturally, you wouldn't have both of the local CLIs in there. When it finds "@quasar/app" it uninstalls it and replaces it with the newer "@quasar/app-webpack" (so that people would have a smooth and automatic transition and not be left on the old q/app 3.3.3). This is exactly what it did for you too.

For the other commands (dev, build, etc), q/cli relinquishes control to the local CLI. It just so happens that it detects the webpack CLI first, before the Vite one.

Your project folder was probably migrated from the webpack variant to the vite one, but in doing so, "@quasar/app" was moved to "dependencies" -- which shouldn't have happened.

Bottom line:
1. Do NOT define both locally installable CLIs in your package.json
2. Do NOT specify the local CLI in "dependencies", but in "devDependencies". This matters quite a lot when building with Electron, SSR or BEX.
